### PR TITLE
add pre & post css transform callbacks to customize css transformatio…

### LINKED ISSUE #145: Support for pre/post transformers similar to webpack loaders
I'm working on a project which builds the app for multiple themes and enables user to switch between them on the fly. We use mixins to set `$selectedTheme = this.$style[theme]` on the component. In the SFC, we do as below:

```
<template>
    <div :class="$selectedTheme['active']">
        Content
    </div>
</template>

<style lang="scss" module themed>
   .active{
      @include active();
   }
</style>
```
We use webpack custom loaders to identify the `themed `attr and build css modules for all the themes.
When we try to test such components, we get error as we are not transforming those css modules.

I know we can mock $styles while we are mounting the components but I believe a **pre & post transform hook** in vue-jest for every type is a **good to have feature** for supporting features  webpack loaders do.

I'm happy to open a PR if it is a valid feature.

### PR DESCRIPTION
**Pull request by yohodopo**
pre & post css transform callbacks to customize css transformation behavior 
Fixes #145,  #125 by allowing users to write custom transformers similar to webpack loaders

### CODE REVIEW

**Comment by vamc-anne** [comment]
Hey @eddyerburgh , I've updated the PR with test cases, can you please look into it. Thanks

File: lib/compilers/helpers/module-name-mapper-helper.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
 const path = require('path')
-
+const matchModuleImport = /^[^?]*~/
 /**
- * Resolves the path to the file locally.
+ * Resolves the path to the file/modeule.
```
Comment (eddyerburgh) [inline lib/compilers/helpers/module-name-mapper-helper.js]:
```suggestion
 * Resolves the path to the file/module.
```

File: lib/compilers/helpers/module-name-mapper-helper.js [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
  *
  * @param {String} to - the name of the file to resolve to
- * @param {String} localPath - the local path
+ * @param {String} importPath - the local path
+ * @param {String} fileType - extn of the file to be resolved
```
Comment (eddyerburgh) [inline lib/compilers/helpers/module-name-mapper-helper.js]:
```suggestion
 * @param {String} fileType - extension of the file to be resolved
```

**Comment by eddyerburgh** [comment]
I agree with this PR, but I think we should make it more general. 

The config option should be `transform`:

```js
{
   jest: { 
    'vue-jest': {
      transform: {
        scss: './custom-transform'
      }
    }
  }
}
```

And each transform should expect to be the same as any Jest transformer:

```js
// custom-transform.js
module.exports = {
  process(src, filePath, config) {
    // custom transform
  }
}
```

We could support a `preProcess` and `postProcess` script that receives the same parameters, to support the functionality you're proposing:

```js
// custom-transform.js
module.exports = {
  process(src, filePath, config) {
    // custom transform
  },
  postProcess(src, filePath, config) {
    // ..
  },
  preProcess(src, filePath, config) {
    // ..
  },
}
```

So in this PR can you make changes to the API to use `transform` as the config option, and expect the transforms to be an object that can contain postProcess and preProcess methods.

**Comment by yohodopo** [comment]
@eddyerburgh, I've updated the PR to make the change generic.

**Comment by eddyerburgh** [comment]
Thanks for the changes, I'm going to give this a full review at the weekend

**Pull request merged**